connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -412,14 +412,13 @@ private void rewind() {
         if (offsets.isEmpty()) {
             return;
         }
-        for (TopicPartition tp: offsets.keySet()) {
-            Long offset = offsets.get(tp);
-            if (offset != null) {
-                log.trace("Rewind {} to offset {}.", tp, offset);
-                consumer.seek(tp, offset);
-                lastCommittedOffsets.put(tp, new OffsetAndMetadata(offset));
-                currentOffsets.put(tp, new OffsetAndMetadata(offset));
-            }
+        for (Map.Entry<TopicPartition, Long> entry: offsets.entrySet()) {
+            TopicPartition tp = entry.getKey();
+            Long offset = entry.getValue();
+            log.trace("Rewind {} to offset {}.", tp, offset);
```

Comment:
I like this cleanup, but I think we still need the `null` check. Since it's possible for the value to be `null`, we should probably be defensive about it. Or were you thinking that we should just let the `NullPointerException` occur and kill the connector? Something in the middle of these two cases might be to log a warning so hopefully the connector developer can fix their code. (The only reason we even need to validate this is due to the `SinkTaskContext.offset(Map<TopicPartition, Long> offsets)` variant, the single partition variant with `long` obviously doesn't have the same issue.)